server-spi-private/src/main/java/org/keycloak/models/utils/AuthenticationFlowResolver.java

```diff
@@ -28,16 +28,19 @@
  */
 public class AuthenticationFlowResolver {
 
+    private static final Logger logger = Logger.getLogger(AuthenticationFlowResolver.class);
+
     public static AuthenticationFlowModel resolveBrowserFlow(AuthenticationSessionModel authSession) {
         AuthenticationFlowModel flow = null;
         ClientModel client = authSession.getClient();
         String clientFlow = client.getAuthenticationFlowBindingOverride(AuthenticationFlowBindings.BROWSER_BINDING);
         if (clientFlow != null) {
             flow = authSession.getRealm().getAuthenticationFlowById(clientFlow);
-            if (flow == null) {
-                throw new ModelException("Client " + client.getClientId() + " has browser flow override, but this flow does not exist");
+            if (flow != null) {
+                return flow;
             }
-            return flow;
+            logger.warnf("Client %s has browser flow override, but this flow '%s' does not exist, " +
```

Comment:
Minor: The `flow` in this logging message will be always null if I am looking correctly? Should the third argument of the `logger.warnf` rather be `clientFlow` variable?